Added application layer for events

leovujanic opened the pull request:
Added event application context

msvrtan commented on src/Event/Behat/EventApplicationContext.php:
```diff
@@ -0,0 +1,148 @@
+     */
+    public function isScheduledFor(string $eventTitle, DateTime $eventDate)
+    {
+        $this->event = new EventEntity(
```
No need to store it as property of context

leovujanic replied:
Fixed

msvrtan commented on src/Event/Handler/RsvpNoHandler.php:
```diff
@@ -0,0 +1,43 @@
+        $event    = $this->eventRepository->load($command->getEventId());
+        $attendee = $this->userRepository->load($command->getUserId());
+
+        $event->addNotComing($attendee);
```
You are forgetting to save it (same mistake I did last Saturday :) )

msvrtan commented on src/Event/Handler/RsvpYesHandler.php:
```diff
@@ -0,0 +1,39 @@
+        $event    = $this->eventRepository->load($command->getEventId());
+        $attendee = $this->userRepository->load($command->getUserId());
+
+        $event->addAttendee($attendee);
```
You are forgetting to save it (same mistake I did last Saturday :) )

leovujanic replied:
Fixed

HcsOmot replied:
@leovujanic I'm not seeing this change - where's the persist/flush call?

flackovic commented on src/Event/Behat/EventApplicationContext.php:
```diff
@@ -0,0 +1,248 @@
+    {
+        $event = $this->eventRepository->loadByTitle($eventTitle);
+
+        Assert::true($event->isAttending($this->currentUser));
```
This asserts that user is *attending* event, but method is called "...ofInterestedMembers"
Do we have distinction between interested and attending? If not, can you rename method in something like "...ofAttendingMembers" ?

msvrtan replied:
Well, interested and attending are not same thing... so maybe the method name should be 'isInterested' instead of 'isAttending' ?

leovujanic commented:
@msvrtan I have squashed remove tag commits. @HcsOmot I have changed commit messages, please check is it ok now. For missing spec, I will open a new issue.

flackovic commented on src/Event/Behat/EventApplicationContext.php:
```diff
@@ -0,0 +1,248 @@
+    public function iWillNotBeOnListOfInterestedMembersForEvent(string $eventTitle)
+    {
+        $event = $this->eventRepository->loadByTitle($eventTitle);
+        Assert::false(in_array($this->currentUser, $event->getAttendees()));
```
Same thing as above.

HcsOmot commented:
hey @leovujanic , what's the status of this? Need any help? I'd be happy to help out any way I can.
:raised_hands:

leovujanic commented:
Hey @HcsOmot ! If you are willing to continue working on it I will be glad. I just can't find enough time right now.

msvrtan commented:
Rebased with master

msvrtan commented:
I finished up this PR, would love if you can all check it out

msvrtan commented:
Rebased with master, there were conflicts on phpstan.neon
